cmd/delete/key/key.go

```diff
@@ -0,0 +1,98 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package key
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+
+	"github.com/ppc64le-cloud/pvsadm/pkg"
+	"github.com/ppc64le-cloud/pvsadm/pkg/client"
+	"github.com/ppc64le-cloud/pvsadm/pkg/utils"
+)
+
+const deletePromptMessage = "Deleting all the above ssh key/key's, Do you really want to continue?"
+
+var searchStr string
+
+var Cmd = &cobra.Command{
+	Use:   "key",
+	Short: "Delete PowerVS ssh key/keys",
+	Long: `Delete PowerVS ssh key/keys
+
+# Set the API key or feed the --api-key commandline argument
+export IBMCLOUD_API_KEY=<IBM_CLOUD_API_KEY>
+
+Examples:
+
+# Delete all the ssh keys which are created before 12hrs
+pvsadm delete key --instance-name upstream-core --before 12h
+
+#Delete all the ssh keys created since 30m
+pvsadm delete key --instance-name upstream-core --since 30m
+
+#Delete all ssh keys starts with rdr-
+pvsadm delete key --instance-name upstream-core --regex "^rdr-.*"
+
+#Delete all ssh keys without asking confirmation
+pvsadm delete key --instance-name upstream-core --noprompt
+
+#Delete all ssh keys and ignore if any errors during the delete operation
+pvsadm delete key --instance-name upstream-core --ignore-errors
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		opt := pkg.Options
+
+		c, err := client.NewClientWithEnv(opt.APIKey, opt.Environment, opt.Debug)
+		if err != nil {
+			klog.Errorf("failed to create a session with IBM cloud: %v", err)
+			return err
+		}
+
+		pvmclient, err := client.NewPVMClientWithEnv(c, opt.InstanceID, opt.InstanceName, opt.Environment)
+		if err != nil {
+			return err
+		}
+
+		keys, err := pvmclient.KeyClient.GetAllPurgeable(pkg.Options.Before, pkg.Options.Since, searchStr)
+		if err != nil {
+			return fmt.Errorf("failed to get the ssh keys, err: %v", err)
+		}
+
+		klog.Infof("keys matched are %s", keys)
+		if len(keys) != 0 {
+			if opt.NoPrompt || utils.AskConfirmation(deletePromptMessage) {
+				for _, key := range keys {
+					err = pvmclient.KeyClient.Delete(key)
+					if err != nil {
+						if opt.IgnoreErrors {
+							klog.Infof("error occurred while deleting the vm: %v", err)
+						} else {
+							return fmt.Errorf("failed to delete a key, err: %v", err)
+						}
+					}
+					klog.Infof("Successfully deleted a key, id: %s", key)
+				}
+			}
+		}
+		return nil
+	},
+}
+
+func init() {
+	Cmd.Flags().StringVar(&searchStr, "regex", "", "Regular Expression to match ssh key/keys")
+}
```

Comment:
we should make regex a mandatory argument for this command, risk of removing someone else's key is high, and let's not allow users to do that.